src/corehost/cli/test/nativehost/comhost_test.cpp

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "comhost_test.h"
+#include <error_codes.h>
+#include <iostream>
+#include <future>
+#include <pal.h>
+
+namespace
+{
+    using DllGetClassObject_fn = HRESULT(STDMETHODCALLTYPE*)(REFCLSID rclsid, REFIID riid, LPVOID *ppv);
```

Comment:
If you are including headers that define `IClassFactory`, I believe that the `DllGetClassObject` declaration should also be available, which means you can use that as the type signature instead of redefining it here.